sqlglot/expressions.py

```diff
@@ -219,10 +220,21 @@ def type(self, dtype: t.Optional[DataType | DataType.Type | str]) -> None:
             dtype = DataType.build(dtype)
         self._type = dtype  # type: ignore
 
+    @property
+    def metadata(self) -> t.Optional[t.Dict[str, t.Any]]:
```

Comment:
rename to just def meta -> t.Dict[str, t.Any]

Reply:
if you're going to keep it as an optional, there's no need for the instance var to be private

not exactly sure how you're planning to use it though, so i think having it default to always a dict is convenient